addons/mass_mailing/models/utm_medium.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models
+
+from odoo.exceptions import UserError
+
+
+class UtmMedium(models.Model):
+    _inherit = 'utm.medium'
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_linked_mailings(self):
```

Comment:
Same here, if it is only for a nice error message, would say that "already managed by restrict but with a nice error message"